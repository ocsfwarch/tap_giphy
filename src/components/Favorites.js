import React from "react";
import ImageDisplay from "./ImageDisplay/ImageDisplay";

function Favorites(props) {
  return (
    <ImageDisplay
      imageList={props.favorites}
      action={props.action}
      icon="times-circle"
      title="Remove From Favorites"
      class="remove"
    />
  );
}

export default Favorites;
