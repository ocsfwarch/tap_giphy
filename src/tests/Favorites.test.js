import React from "react";
import renderer from "react-test-renderer";
import Favorites from "../components/Favorites";

const favorites = [
  {
    id: "26gsvA7sDa86fDu6s",
    image:
      "https://media4.giphy.com/media/26gsvA7sDa86fDu6s/giphy.gif?cid=4647a4e059uevh59ry19ypaw55lv68zntb1h4pmvoujvkr2a&rid=giphy.gif&ct=g",
    desc: "Busch Beer GIF by Busch",
  },
  {
    id: "78ny3pDP74gDu",
    image:
      "https://media4.giphy.com/media/78ny3pDP74gDu/giphy.gif?cid=4647a4e059uevh59ry19ypaw55lv68zntb1h4pmvoujvkr2a&rid=giphy.gif&ct=g",
    desc: "vs hunter GIF",
  },
  {
    id: "K9b2WiPZi0ZjO",
    image:
      "https://media4.giphy.com/media/K9b2WiPZi0ZjO/giphy.gif?cid=4647a4e059uevh59ry19ypaw55lv68zntb1h4pmvoujvkr2a&rid=giphy.gif&ct=g",
    desc: "deer popcorn GIF",
  },
];

it("renders correctly", () => {
  const tree = renderer.create(<Favorites favorites={favorites} />).toJSON();
  expect(tree).toMatchSnapshot();
});
