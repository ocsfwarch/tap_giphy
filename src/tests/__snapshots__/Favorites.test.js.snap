// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="image_display_container"
>
  <section
    className="image_list"
  >
    <ul
      className="hide_list"
    />
  </section>
</div>
`;
